subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectFactory.java

```diff
@@ -44,6 +47,10 @@ public ProjectFactory(Instantiator instantiator, TextFileResourceLoader textFile
 
     @Override
     public ProjectInternal createProject(GradleInternal gradle, ProjectDescriptor projectDescriptor, ProjectState owner, @Nullable ProjectInternal parent, ServiceRegistryFactory serviceRegistryFactory, ClassLoaderScope selfClassLoaderScope, ClassLoaderScope baseClassLoaderScope) {
+        if (isRootDir(projectDescriptor)) {
```

Comment:
This code seems like its in the wrong place. IMO this belongs in where the the actual project initialization is taking place, in [InitBuild.java](https://github.com/gradle/gradle/blob/master/subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java#L182)